cmd/fs-v1-multipart_test.go

```diff
@@ -20,15 +20,76 @@ import (
 	"bytes"
 	"path/filepath"
 	"testing"
+	"time"
 )
 
+func TestFSCleanupMultipartUploadsInRoutine(t *testing.T) {
+	// Prepare for tests
+	disk := filepath.Join(globalTestTmpDir, "minio-"+nextSuffix())
+	defer removeAll(disk)
+
+	obj := initFSObjects(disk, t)
+	fs := obj.(*fsObjects)
+
+	bucketName := "bucket"
+	objectName := "object"
+
+	obj.MakeBucketWithLocation(bucketName, "")
+	_, err := obj.NewMultipartUpload(bucketName, objectName, nil)
+	if err != nil {
+		t.Fatal("Unexpected err: ", err)
+	}
+
+	// Close the current active go-routine.
+	close(fs.staleMultipartUploadsDoneCh)
```

Comment:
Close the (active) go-routine before NewMultipartUpload request  to ensure that the new routine is the one that performed the clean up.